receiver/splunkhecreceiver/receiver.go

```diff
@@ -231,20 +253,75 @@ func (r *splunkReceiver) Shutdown(context.Context) error {
 	return err
 }
 
-func (r *splunkReceiver) writeSuccessResponse(ctx context.Context, resp http.ResponseWriter, eventCount int) {
+func (r *splunkReceiver) writeSuccessResponseWithAck(ctx context.Context, resp http.ResponseWriter, eventCount int, partitionID string) {
+	ackID := r.ackExt.ProcessEvent(partitionID)
+	r.ackExt.Ack(partitionID, ackID)
+	r.writeSuccessResponse(ctx, resp, eventCount, []byte(fmt.Sprintf(responseOKWithAckID, ackID)))
+}
+
+func (r *splunkReceiver) writeSuccessResponseWithoutAck(ctx context.Context, resp http.ResponseWriter, eventCount int) {
```

Comment:
Do we need this function? Is there a reason we wouldn't directly call `writeSuccessResponse`?

Reply:
We could. The value this function adds is not requiring the caller to provide one additional parameter (the response body content). By having this intermediate function I thought there would be a reduced likelyhood of mistakes/misuse from the caller.

Reply:
I'd probably prefer to remove this method, but just a nit. Your call.

Reply:
removed